freqtrade/rpc/api_server/ws/serializer.py

```diff
@@ -0,0 +1,89 @@
+import json
+import logging
+from abc import ABC, abstractmethod
+
+import msgpack
+import orjson
+import rapidjson
+from pandas import DataFrame
+
+from freqtrade.misc import dataframe_to_json, json_to_dataframe
+from freqtrade.rpc.api_server.ws.proxy import WebSocketProxy
+
+
+logger = logging.getLogger(__name__)
+
+
+class WebSocketSerializer(ABC):
+    def __init__(self, websocket: WebSocketProxy):
+        self._websocket: WebSocketProxy = websocket
+
+    @abstractmethod
+    def _serialize(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def _deserialize(self, data):
+        raise NotImplementedError()
+
+    async def send(self, data: bytes):
+        await self._websocket.send(self._serialize(data))
+
+    async def recv(self) -> bytes:
+        data = await self._websocket.recv()
+
+        return self._deserialize(data)
+
+    async def close(self, code: int = 1000):
+        await self._websocket.close(code)
+
+
+class JSONWebSocketSerializer(WebSocketSerializer):
+    def _serialize(self, data):
+        return json.dumps(data, default=_json_default)
```

Comment:
The `_json_default` is a very dangerous approach to use for a serializer.
it'll require you to implement ALL possible types.

if i subscribe to exit_fill, i get for example the following, as serialization for datetime is not implemented.


```
2022-09-05 19:42:52,689 - freqtrade.rpc.api_server.webserver - ERROR - Exception happened in background task: Type is not JSON serializable: datetime.timedelta
Traceback (most recent call last):
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/webserver.py", line 198, in _broadcast_queue_data
    await self._channel_manager.broadcast(message)
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/ws/channel.py", line 159, in broadcast
    await channel.send(data)
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/ws/channel.py", line 54, in send
    await self._wrapped_ws.send(data)
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/ws/serializer.py", line 30, in send
    await self._websocket.send(self._serialize(data))
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/ws/serializer.py", line 61, in _serialize
    return orjson.dumps(data, default=_json_default)
TypeError: Type is not JSON serializable: datetime.timedelta
2022-09-05 19:42:52,692 - freqtrade.wallets - INFO - Wallets synced.
```